src/backend/asset-type-tree/assetTypeTreeOps.ts

```diff
@@ -0,0 +1,400 @@
+    options = options as AddOrRemoveChildOptions;
+    return options.CHILD_ID !== undefined && options.PARENT_ID != undefined;
+}
+
+export type MethodOptions = CreateAssetTypeOptions | DeleteAssetTypeOptions | AddOrRemoveChildOptions;
+
+export interface AssetTypeTreeOptions {
+    METHOD_NAME: AssetTypeTreeMethod;
+    METHOD_OPTIONS: MethodOptions;
+}
+
+export function assetTypeTreeHandler(req: CbServer.BasicReq, resp: CbServer.Resp, options: AssetTypeTreeOptions): void {
+    const assetTypeTreeCollection = ClearBlade.Collection({ collectionName: CollectionName.ASSET_TYPE_TREE });
+    const getTreeQuery = ClearBlade.Query().setPage(1, 1);
+
+    const callback = (err: any, data: any) => {
+        if (err) {
+            resp.error('Error: ' + err);
+        } else {
+            const itemID = data.DATA[0]['item_id'];
+            const treeStr = data.DATA[0]['tree'];
+            const assetTypeTree = new AssetTypeTree(itemID, req, resp, AssetTypeTree.treeFromString(treeStr));
+
+            switch (options.METHOD_NAME) {
+                case AssetTypeTreeMethod.GET_TREE:
+                    resp.success(assetTypeTree.getTree());
+                    break;
+                case AssetTypeTreeMethod.GET_TOP_LEVEL_ASSET_TYPES:
+                    resp.success(assetTypeTree.getTopLevelAssetTypes());
+                    break;
+                case AssetTypeTreeMethod.CREATE_ASSET_TYPE:
+                    if (isCreateAssetTypeOptions(options.METHOD_OPTIONS)) {
+                        assetTypeTree.createAssetType(options.METHOD_OPTIONS);
+                    } else {
+                        resp.error('Error: Missing CreateAssetTypeOptions.');
+                    }
+                    break;
+                case AssetTypeTreeMethod.DELETE_ASSET_TYPE:
+                    if (isDeleteAssetTypeOptions(options.METHOD_OPTIONS)) {
+                        assetTypeTree.deleteAssetType(options.METHOD_OPTIONS);
+                    } else {
+                        resp.error('Error: Missing DeleteAssetTypeOptions.');
+                    }
+                    break;
+                case AssetTypeTreeMethod.REMOVE_CHILD:
+                    if (isAddOrRemoveChildOptions(options.METHOD_OPTIONS)) {
+                        assetTypeTree.removeChild(options.METHOD_OPTIONS);
+                    } else {
+                        resp.error('Error: Missing AddOrRemoveChildOptions.');
+                    }
+                    break;
+                case AssetTypeTreeMethod.ADD_CHILD:
+                    if (isAddOrRemoveChildOptions(options.METHOD_OPTIONS)) {
+                        assetTypeTree.addChild(options.METHOD_OPTIONS);
+                    } else {
+                        resp.error('Error: Missing AddOrRemoveChildOptions.');
+                    }
+                    break;
+                default:
+                    break;
+            }
+            resp.success(AssetTypeTree.treeToString(assetTypeTree.nodes));
```

Comment:
call a callback within individual cases using promises and get rid of the statement.
Throw an error in default with "method not supported or not found"